src/objectid.ts

```diff
@@ -56,18 +54,20 @@ export class ObjectId {
   /**
    * Create an ObjectId type
    *
-   * @param {(string|Buffer|number)} id Can be a 24 byte hex string, 12 byte binary Buffer, or a Number.
-   * @property {number} generationTime The generation time of this ObjectId instance
-   * @return {ObjectId} instance of ObjectId.
+   * @param id - Can be a 24 character hex string, 12 byte binary Buffer, or a number.
    */
-  constructor(id?: string | Buffer | number) {
+  constructor(id?: string | Buffer | number | ObjectIdLike | ObjectId) {
     // Duck-typing to support ObjectId from different npm packages
-    if (id instanceof ObjectId) return id;
+    if (id instanceof ObjectId) {
+      this.id = id.id;
+      this.__id = id.__id;
+      return;
+    }
 
     // The most common use case (blank id, new objectId instance)
     if (id == null || typeof id === 'number') {
       // Generate a new id
-      this.id = ObjectId.generate(id);
+      this.id = ObjectId.generate(typeof id === 'number' ? id : undefined);
```

Comment:
now I see where all the checks for `string` are coming from. Let's solve this once and for all, the type for `id` should just be a `Buffer`

Reply:
This was lost as well, we can do it in follow-on work but please make a ticket for that

Reply:
NODE-2821